### PR TITLE
[functorch] add linalg cross batch rule

### PR DESCRIPTION
**Pull request by samdow**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83761
* #83760
* #82325
* __->__ #83759

### CODE REVIEW

File: functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp [hunk 1]
```diff
@@ -335,6 +335,22 @@ oneOutput matrix_exp_batch_rule(const Tensor& self, c10::optional<int64_t> self_
   return std::make_tuple(std::get<0>(res), 0, std::get<1>(res), 0, std::get<2>(res), 0, std::get<3>(res), 0);
 }
 
+oneOutput cross_batch_rule(const Tensor& self, c10::optional<int64_t> self_bdim,
+                           const Tensor& other, c10::optional<int64_t> other_bdim, const int64_t dim) {
+  const auto batch_size = get_bdim_size2(self, self_bdim, other, other_bdim);
+
+  const auto self_other_bundled = _binary_pointwise_helper(self, self_bdim, other, other_bdim, false);
+
+  // might be a bug but still doesn't incur an extra perf hit because input would be expanded in cross' broadcast
```
Comment (samdow) [inline functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp:344]: filed #83756 to see if this is expected behavior since cross does broadcast its inputs

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**